numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -998,18 +998,33 @@ VOID_setitem(PyObject *op, void *input, void *vap)
      * undiscerning case: It interprets any object as a buffer
      * and reads as many bytes as possible, padding with 0.
      */
+#if defined(NPY_PY3K)
+    {
+        Py_buffer view;
+
+        if (PyObject_GetBuffer(op, &view, PyBUF_SIMPLE) < 0) {
+            return -1;
+        }
+        memcpy(ip, view.buf, PyArray_MIN(view.len, itemsize));
```

Comment:
I'm a little worried about the meaning of `len` here. Can you add a test for setting a `(np.void, 7)` with a `ctypes.c_uint16.__ctype_le__ * 2 * 2`?

Reply:
I believe it is the same as it was before, just differently expressed. I'm going off the Python3.7 implementation of the previous functions.  What are your concerns?

Reply:
I incorrectly assumed that `len` was `shape[0]`. It would still be nice to add the ctypes test, but I guess out of scope.